subprojects/core/src/main/java/org/gradle/plugin/repository/IvyPluginRepository.java

```diff
@@ -44,4 +47,103 @@
+     */
+    void ivyPattern(String pattern);
+
+    /**
+     * Specifies the layout to use with this repository, based on the root url.
+     * See {@link #layout(String, Closure)}.
+     *
+     * @param layoutName The name of the layout to use.
+     */
+    void layout(String layoutName);
+
+    /**
+     * Specifies how the items of the repository are organized.
+     * <p>
+     * The layout is configured with the supplied closure.
+     * <p>
+     * Recognised values are as follows:
+     * </p>
+     * <h4>'gradle'</h4>
+     * <p>
+     * A Repository Layout that applies the following patterns:
+     * <ul>
+     *     <li>Artifacts: <code>$baseUri/{@value #GRADLE_ARTIFACT_PATTERN}</code></li>
+     *     <li>Ivy: <code>$baseUri/{@value #GRADLE_IVY_PATTERN}</code></li>
+     * </ul>
+     * </p>
+     * <h4>'maven'</h4>
+     * <p>
+     * A Repository Layout that applies the following patterns:
+     * <ul>
+     *     <li>Artifacts: <code>$baseUri/{@value #MAVEN_ARTIFACT_PATTERN}</code></li>
+     *     <li>Ivy: <code>$baseUri/{@value #MAVEN_IVY_PATTERN}</code></li>
+     * </ul>
+     * </p>
+     * <p>
+     * Following the Maven convention, the 'organisation' value is further processed by replacing '.' with '/'.
+     * </p>
+     * <h4>'ivy'</h4>
+     * <p>
+     * A Repository Layout that applies the following patterns:
+     * <ul>
+     *     <li>Artifacts: <code>$baseUri/{@value #IVY_ARTIFACT_PATTERN}</code></li>
+     *     <li>Ivy: <code>$baseUri/{@value #IVY_ARTIFACT_PATTERN}</code></li>
+     * </ul>
+     * </p>
+     * <p><b>Note:</b> this pattern is currently {@link org.gradle.api.Incubating incubating}.</p>
+     * <h4>'pattern'</h4>
+     * <p>
+     * A repository layout that allows custom patterns to be defined. eg:
+     * <pre autoTested="">
+     * repositories {
+     *     ivy {
+     *         layout 'pattern' , {
+     *             artifact '[module]/[revision]/[artifact](.[ext])'
+     *             ivy '[module]/[revision]/ivy.xml'
+     *         }
+     *     }
+     * }
+     * </pre>
+     * </p>
+     * <p>The available pattern tokens are listed as part of <a href="http://ant.apache.org/ivy/history/latest-milestone/concept.html#patterns">Ivy's Main Concepts documentation</a>.</p>
+     *
+     * @param layoutName The name of the layout to use.
+     * @param config The action used to configure the layout.
+     * @since 2.3 (feature was already present in Groovy DSL, this particular method introduced in 2.3)
+     */
+    void layout(String layoutName, Action<? extends RepositoryLayout> config);
+
+    /**
+     * Specifies how the items of the repository are organized. See {@link #layout(String, org.gradle.api.Action)}
+     *
+     * @param layoutName The name of the layout to use.
+     * @param config The closure used to configure the layout.
+     * An instance of {@link RepositoryLayout} is passed as a parameter to the closure.
+     */
+    void layout(String layoutName, Closure config);
```

Comment:
I can remove it, do you want me to also remove it from the RepositoryHandler that it was copied from?